src/certs/snp/chain.rs

```diff
@@ -97,32 +105,39 @@ impl Chain {
             return Err(Error::new(other, "ARK not found"));
         } else if ask.is_none() {
             return Err(Error::new(other, "ASK not found"));
-        } else if vcek.is_none() {
-            return Err(Error::new(other, "VCEK not found"));
+        } else if vcek.is_none() && vlek.is_none() {
+            return Err(Error::new(other, "VCEK/VLEK not found"));
         }
 
+        // Use the VLEK whenever it is present, but use the VCEK when VLEK is missing.
+        let vek_val = match (vcek, vlek) {
+            (_, Some(vlek)) => vlek,
+            (Some(vcek), None) => vcek,
+            _ => unreachable!(),
```

Comment:
`(Some(vcek), Some(vlek)`?

Reply:
That is covered on line 114

Reply:
I misread what that meant. Thanks! LGTM.